src/engine/build.sh

```diff
@@ -124,7 +124,10 @@ guess_toolset ()
     elif test_uname FreeBSD && test_path freebsd-version && test_path clang++ && test_cxx11 clang ; then B2_TOOLSET=clang
     elif test_path g++ && test_cxx11 gcc ; then B2_TOOLSET=gcc
     elif test_path clang++ && test_cxx11 clang ; then B2_TOOLSET=clang
-    elif test_path icc && test_cxx11 intel-linux ; then B2_TOOLSET=intel-linux
+    elif test_path icpc && test_cxx11 intel-linux ; then B2_TOOLSET=intel-linux
+    elif test -r /opt/intel/inteloneapi/setvars.sh && test_cxx11 intel-linux ; then
+        B2_TOOLSET=intel-linux
+        B2_TOOLSET_ROOT=/opt/intel/inteloneapi
```

Comment:
Looking at the logic below, I think this should read:
```suggestion
        B2_TOOLSET_ROOT=/opt/intel/inteloneapi/compiler/latest/linux/
```

Reply:
Unfortunately, the `iccvars.sh` script in `bin/` seems to be a dead link on my local copy of oneAPI pulled via the apt repos: https://software.intel.com/content/www/us/en/develop/articles/oneapi-repo-instructions.html

We should instead source `/opt/intel/inteloneapi/setvars.sh` below with this one.

Reply:
I do not have that distribution right now. So if anyone with that distribution can confirm I'll commit the suggestion.

Reply:
I'll post a suggestion, give me a sec :)

Reply:
@aminiussi can you please mark this one as "resolved"? :)